File: BraintreeCore/src/main/java/com/braintreepayments/api/VenmoConfiguration.java
package com.braintreepayments.api;

import android.text.TextUtils;

import org.json.JSONObject;

/**
 * Contains the remote Venmo configuration for the Braintree SDK.
 */
class VenmoConfiguration {

    private static final String ACCESS_TOKEN_KEY = "accessToken";
    private static final String ENVIRONMENT_KEY = "environment";
    private static final String MERCHANT_ID_KEY = "merchantId";

    private String accessToken;
    private String environment;
    private String merchantId;

    /**
     * Parses the Venmo configuration from json.
     *
     * @param json The json to parse.
     * @return A {@link VenmoConfiguration} instance with data that was able to be parsed from the
     * {@link JSONObject}.
     */
    static VenmoConfiguration fromJson(JSONObject json) {
        if (json == null) {
            json = new JSONObject();
        }

        VenmoConfiguration venmoConfiguration = new VenmoConfiguration();
        venmoConfiguration.accessToken = Json.optString(json, ACCESS_TOKEN_KEY, "");
        venmoConfiguration.environment = Json.optString(json, ENVIRONMENT_KEY, "");
        venmoConfiguration.merchantId = Json.optString(json, MERCHANT_ID_KEY, "");

        return venmoConfiguration;
    }

    /**
     * @return The access token to use with Venmo.
     */
    String getAccessToken() {
        return accessToken;
    }

    /**
     * @return The merchant Id associated with this merchant's Venmo integration.
     */
    String getMerchantId() {
        return merchantId;
    }

    /**
     * @return The Venmo environment the merchant is running in.
     */
    String getEnvironment() {
        return environment;
    }

    boolean isAccessTokenValid() {
        return !TextUtils.isEmpty(accessToken);
    }
}

File: BraintreeCore/src/main/java/com/braintreepayments/api/VenmoConfiguration.kt
package com.braintreepayments.api

import android.text.TextUtils
import org.json.JSONObject

/**
 * Contains the remote Venmo configuration for the Braintree SDK.
 * @property accessToken The access token to use with Venmo.
 * @property environment The Venmo environment the merchant is running in.
 * @property merchantId The merchant Id associated with this merchant's Venmo integration.
 * @property enrichedCustomerDataEnabled The boolean value indicating whether ECD is enabled for the merchant.
 */
internal class VenmoConfiguration constructor(
    val accessToken: String,
    val environment: String,
    val merchantId: String,
    val enrichedCustomerDataEnabled: Boolean
) {

    constructor(json: JSONObject?) : this(
        Json.optString(json, ACCESS_TOKEN_KEY, ""),
        Json.optString(json, ENVIRONMENT_KEY, ""),
        Json.optString(json, MERCHANT_ID_KEY, ""),
        Json.optBoolean(json, ECD_ENABLED_KEY, false),
    )

    val isAccessTokenValid: Boolean = !TextUtils.isEmpty(accessToken)

    companion object {
        private const val ACCESS_TOKEN_KEY = "accessToken"
        private const val ENVIRONMENT_KEY = "environment"
        private const val MERCHANT_ID_KEY = "merchantId"
        private const val ECD_ENABLED_KEY = "enrichedCustomerDataEnabled"
    }
}
